packages/itwin/ec3-widget/src/components/CreateAssemblyComponent.tsx

```diff
@@ -0,0 +1,344 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { Button, ButtonGroup, ExpandableBlock, IconButton, Label, LabeledInput, LabeledSelect, ProgressRadial, Select } from "@itwin/itwinui-react";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import type { Configuration } from "./EC3/Template";
+import React from "react";
+import type { EC3ConfigurationLabel, GroupMinimal, ODataTable } from "@itwin/insights-client";
+import "./CreateAssemblyComponent.scss";
+import { SvgAdd, SvgDelete, SvgEdit } from "@itwin/itwinui-icons-react";
+import { useApiContext } from "./context/APIContext";
+
+export interface CreateAssemblyProps {
+  template: Configuration;
+  onNextClick: () => void;
+  onCancelClick?: () => void;
+  onBackClick: () => void;
+  setTemplate: (template: Configuration) => void;
+  label?: EC3ConfigurationLabel[];
+}
+
+export enum CreateAssemblyDropdownType {
+  material,
+  elementName,
+  elementQuantity,
+}
+
+export const CreateAssembly = (props: CreateAssemblyProps) => {
+  const [allAssemblies, setAllAssemblies] = useState<EC3ConfigurationLabel[] | undefined>(props.label);
+  const [reportTables, setReportTables] = useState<string[] | undefined>(undefined);
+  const [groups, setGroups] = useState<GroupMinimal[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [oDataTable, setoDataTable] = useState<ODataTable[]>([]);
+  const [editableAssemblyIndex, setEditableAssemblyIndex] = useState<number>();
+
+  const oDataClient = useApiContext().oDataClient;
+  const {
+    config: { getAccessToken, defaultMapping },
+  } = useApiContext();
+  const groupsClient = useApiContext().groupsClient;
+
+  useMemo(() => {
+    setAllAssemblies(props.label);
+  }, [props.label]);
+
+  const getGroups = useCallback(async () => {
```

Comment:
You should not be using `Groups` from the `Grouping & Mapping` API in any way in this widget. You should only be using whatever is in the Report's OData feed.

To get the metadata of the report, you can use `IOdataClient.getODataReportMetadata`. It returns a list of tables (groups) and its columns (properties).

Reply:
`IOdataClient` is in @itwin/insights-client (https://github.com/iTwin/insights-client/blob/main/src/reporting/clients/IODataClient.ts).

Reply:
Removed all code related to grouping